tls/s2n_handshake.c

```diff
@@ -70,3 +70,35 @@ int s2n_handshake_parse_header(struct s2n_connection *conn, uint8_t * message_ty
 
     return 0;
 }
+
+int s2n_handshake_get_hash_state(struct s2n_connection *conn, s2n_hash_algorithm hash_alg, struct s2n_hash_state *hash_state)
+{
+    switch (hash_alg) {
+    case S2N_HASH_MD5:
+        *hash_state = conn->handshake.md5;
```

Comment:
I'm not sure why we need to copy the state instead of just returning a reference. But this should work. This will probably break if the internal hash state changes to use opaque pointers.